DigitalLearningSolutions.Web/Helpers/SupervisorCompetencyFilterHelper.cs

```diff
@@ -0,0 +1,129 @@
+        {
+            var filteredCompetencies = competencies;
+            if (search != null)
+            {
+                var searchText = search.SearchText?.Trim() ?? string.Empty;
+                var filters = search.AppliedFilters?.Select(f => int.Parse(f.FilterValue)) ?? Enumerable.Empty<int>();
+                search.CompetencyFlags = competencyFlags.ToList();
+                ApplyResponseStatusFilters(ref filteredCompetencies, filters, searchText);
+                UpdateRequirementsFilterDropdownOptionsVisibility(search, filteredCompetencies);
+                ApplyRequirementsFilters(ref filteredCompetencies, filters);
+
+                foreach (var competency in filteredCompetencies)
+                    competency.CompetencyFlags = search.CompetencyFlags.Where(f => f.CompetencyId == competency.Id);
+
+                ApplyCompetencyGroupFilters(ref filteredCompetencies, search);
+            }
+            return filteredCompetencies;
+        }
+
+        private static void ApplyResponseStatusFilters(ref IEnumerable<Competency> competencies, IEnumerable<int> filters, string searchText = "")
+        {
+            var filteredCompetencies = competencies;
+            var appliedResponseStatusFilters = filters.Where(f => IsResponseStatusFilter(f));
+            if (appliedResponseStatusFilters.Any() || searchText.Length > 0)
+            {
+                var wordsInSearchText = searchText.Split().Where(w => w != string.Empty);
+                filters = appliedResponseStatusFilters;
+                filteredCompetencies = from c in competencies
+                                       let searchTextMatchesGroup = wordsInSearchText.Any(w => c.CompetencyGroup?.Contains(w, StringComparison.CurrentCultureIgnoreCase) ?? false)
+                                       let searchTextMatchesCompetencyDescription = wordsInSearchText.Any(w => c.Description?.Contains(w, StringComparison.CurrentCultureIgnoreCase) ?? false)
+                                       let searchTextMatchesCompetencyName = wordsInSearchText.Any(w => c.Name?.Contains(w, StringComparison.CurrentCultureIgnoreCase) ?? false)
+                                       let responseStatusFilterMatchesAnyQuestion =
+                                          (filters.Contains((int)SelfAssessmentCompetencyFilter.RequiresSelfAssessment) && c.AssessmentQuestions.Any(q => q.ResultId == null))
+                                       || (filters.Contains((int)SelfAssessmentCompetencyFilter.SelfAssessed) && c.AssessmentQuestions.Any(q => q.ResultId != null && q.Requested == null && q.SignedOff == null))
+                                       || (filters.Contains((int)SelfAssessmentCompetencyFilter.PendingConfirmation) && c.AssessmentQuestions.Any(q => q.Verified == null && q.Requested != null))
```

Comment:
It's this logic that doesn't appear to be working correctly - cross reference it with reference to the logic for the tags.